### ISSUE TITLE
Error running event-replay during upgrade from 4.03->5.0.1

### ISSUE THREAD

**Issue by wileyj** [hirosystems/stacks-blockchain-api#1336]
During an upgrade to the API, I ran into an error using a process that previously worked for running an event-replay `import`. 
Postgres version: 14
Postgres data was created/running with API version 4.0.3
tsv file exists and appears to be correct (multiple GB file with expected contents)

1. All services were stopped normally
2. Postgres was brought back up, confirmed to be running and accepting connections
3.  API v5.0.1 was started via docker with `/bin/sh` to run the import manually

the command run to import the data: `node ./lib/index.js import-events --file /tmp/event-replay/stacks-node-events.tsv --wipe-db --force`, which is the same command that has worked with previous upgrades

```
/app # node ./lib/index.js import-events --file /tmp/event-replay/stacks-node-events.tsv  --wipe-db --force
{"level":"error","message":"Error executing:\nCREATE MATERIALIZED VIEW \"mempool_digest\" AS \n      SELECT COALESCE(to_hex(bit_xor(tx_short_id)), '0') AS digest\n      FROM (\n        SELECT ('x' || encode(tx_id, 'hex'))::bit(64)::bigint tx_short_id\n        FROM mempool_txs\n        WHERE pruned = false\n      ) m\n    ;\nerror: relation \"mempool_digest\" already exists\n","timestamp":"2022-09-29T22:06:05.818Z"}
{"code":"42P07","file":"createas.c","length":109,"level":"error","line":"406","message":"Error running pg-migrate relation \"mempool_digest\" already exists","name":"error","routine":"CreateTableAsRelExists","severity":"ERROR","stack":"error: relation \"mempool_digest\" already exists\n    at Parser.parseErrorMessage (/app/node_modules/pg-protocol/src/parser.ts:369:69)\n    at Parser.handlePacket (/app/node_modules/pg-protocol/src/parser.ts:188:21)\n    at Parser.parse (/app/node_modules/pg-protocol/src/parser.ts:103:30)\n    at Socket.<anonymous> (/app/node_modules/pg-protocol/src/index.ts:7:48)\n    at Socket.emit (node:events:513:28)\n    at addChunk (node:internal/streams/readable:315:12)\n    at readableAddChunk (node:internal/streams/readable:289:9)\n    at Socket.Readable.push (node:internal/streams/readable:228:10)\n    at TCP.onStreamRead (node:internal/stream_base_commons:190:23)","timestamp":"2022-09-29T22:06:05.843Z"}
error: relation "mempool_digest" already exists
    at Parser.parseErrorMessage (/app/node_modules/pg-protocol/src/parser.ts:369:69)
    at Parser.handlePacket (/app/node_modules/pg-protocol/src/parser.ts:188:21)
    at Parser.parse (/app/node_modules/pg-protocol/src/parser.ts:103:30)
    at Socket.<anonymous> (/app/node_modules/pg-protocol/src/index.ts:7:48)
    at Socket.emit (node:events:513:28)
    at addChunk (node:internal/streams/readable:315:12)
    at readableAddChunk (node:internal/streams/readable:289:9)
    at Socket.Readable.push (node:internal/streams/readable:228:10)
    at TCP.onStreamRead (node:internal/stream_base_commons:190:23) {
  length: 109,
  severity: 'ERROR',
  code: '42P07',
  detail: undefined,
  hint: undefined,
  position: undefined,
  internalPosition: undefined,
  internalQuery: undefined,
  where: undefined,
  schema: undefined,
  table: undefined,
  column: undefined,
  dataType: undefined,
  constraint: undefined,
  file: 'createas.c',
  line: '406',
  routine: 'CreateTableAsRelExists'
}
```

The next test i ran was to first wipe the postgres DB, then re-attempt the import using the tsv file:
```

{"level":"info","message":"Eventobserverlisteningat:http://127.0.0.1:36707","timestamp":"2022-09-29T22:30:38.989Z"}
{"level":"info","message":"Stacks1.0tokenofferingdataimportstarted","timestamp":"2022-09-29T22:30:39.027Z"}
{"level":"info","message":"Stacks1.0tokenofferingdataimportcompleted","timestamp":"2022-09-29T22:30:40.527Z"}
{"level":"warn","message":"Notice:fullBNSfunctionalityrequires'BNS_IMPORT_DIR'tobeset.","timestamp":"2022-09-29T22:30:40.528Z"}
Importingraweventrequestsintotemporarytable...
error:extradataafterlastexpectedcolumn
atParser.parseErrorMessage(/app/node_modules/pg-protocol/src/parser.ts:369:69)
atParser.handlePacket(/app/node_modules/pg-protocol/src/parser.ts:188:21)
atParser.parse(/app/node_modules/pg-protocol/src/parser.ts:103:30)
atSocket.<anonymous>(/app/node_modules/pg-protocol/src/index.ts:7:48)
atSocket.emit(node:events:513:28)
ataddChunk(node:internal/streams/readable:315:12)
atreadableAddChunk(node:internal/streams/readable:289:9)
atSocket.Readable.push(node:internal/streams/readable:228:10)
atTCP.onStreamRead(node:internal/stream_base_commons:190:23){
    length:240,
    severity:'ERROR',
    code:'22P04',
    detail:undefined,
    hint:undefined,
    position:undefined,
    internalPosition:undefined,
    internalQuery:undefined,
    where:'COPYtemp_raw_tsv,line470545:"470693\t2022-08-2014:13:14.982328+00\t/new_block\t{"events":[{"txid":"0x5559844aefb5d7e7e28d45dc05ea..."',
    schema:undefined,
    table:undefined,
    column:undefined,
    dataType:undefined,
    constraint:undefined,
    file:'copyfromparse.c',
    |ine:'839',
    routine:'NextCopyFrom'
}
import-eventsprocessfailed
```

and the postgres logs from the import:
```

2022-09-2922:30:36.342UTC[1]LOG:startingQL14.5onx86_64-pc-linux-musl,compiledbygcc(Alpine11.2.1_git20220219)11.2.120220219,64-bit
2022-09-2922:30:36.342UTC[1]LOG:listeningonIPv4address"0.0.0.0",port5432
2022-09-2922:30:36.342UTC[1]LOG:listeningonIPv6address"::",port5432
2022-09-2922:30:36.347UTC[1]LOG:listeningonUnixsocket"/var/run/ql/.s.PGSQL.5432"
2022-09-2922:30:36.353UTC[48]LOG:databasesystemwasshutdownat2022-09-2922:30:36UTC
2022-09-2922:30:36.358UTC[1]LOG:databasesystemisreadytoacceptconnections
2022-09-2922:30:36.602UTC[56]ERROR:relation"event_observer_requests"doesnotexistatcharacter16
2022-09-2922:30:36.602UTC[56]STATEMENT:SELECTidfromevent_observer_requestsLIMIT1
2022-09-2922:36:27.442UTC[63]ERROR:extradataafterlastexpectedcolumn
2022-09-2922:36:27.442UTC[63]CONTEXT:COPYtemp_raw_tsv,line470545:"470693	2022-08-2014:13:14.982328+00	/new_block	{"events":[{"txid":"0x5559844aefb5d7e7e28d45dc05ea..."
2022-09-2922:36:27.442UTC[63]STATEMENT:COPYtemp_raw_tsvFROMSTDIN
```

Based on this, i suspected that perhaps the tsv file may be corrupt, so the next test was to restore the postges DB  from  a backup taken before the import test (again, postgres 14 data from API v4.0.3). 
However, running the `export` results in a different (unlogged) error:
```
Exporteventdatatofile:/tmp/event-replay/stacks-node-events.tsv
Exportstarted...
Exportsuccessful.
{"level":"info","message":"Shuttingdown...receivedbeforeExit.","timestamp":"2022-09-29T22:42:25.619Z"}
{"level":"info","message":"Appshutdownsuccessful.","timestamp":"2022-09-29T22:42:25.621Z"}
```

the result here is an empty tsv file, but the DB seems fine.

**Comment by wileyj** [comment]
Edit: i attempted the export using 5.0.1, not 4.0.3 which created the data.

**Comment by wileyj** [comment]
for now, it seems only syncing from genesis is working - i've tried again using a snapshot of the data, with a verified tsv file (confirmed working with 4.0.3 for import). same errors (adding error again since above is malformed):
```
Event file's block height: 74187
Starting event import and playback in archival mode
{"level":"info","message":"Event observer listening at: http://127.0.0.1:46061","timestamp":"2022-09-30T03:00:18.601Z"}
{"level":"info","message":"Stacks 1.0 token offering data import started","timestamp":"2022-09-30T03:00:18.635Z"}
{"level":"info","message":"Stacks 1.0 token offering data import completed","timestamp":"2022-09-30T03:00:20.105Z"}
{"level":"warn","message":"Notice: full BNS functionality requires 'BNS_IMPORT_DIR' to be set.","timestamp":"2022-09-30T03:00:20.106Z"}
Importing raw event requests into temporary table...
error: extra data after last expected column
    at Parser.parseErrorMessage (/app/node_modules/pg-protocol/src/parser.ts:369:69)
    at Parser.handlePacket (/app/node_modules/pg-protocol/src/parser.ts:188:21)
    at Parser.parse (/app/node_modules/pg-protocol/src/parser.ts:103:30)
    at Socket.<anonymous> (/app/node_modules/pg-protocol/src/index.ts:7:48)
    at Socket.emit (node:events:513:28)
    at addChunk (node:internal/streams/readable:315:12)
    at readableAddChunk (node:internal/streams/readable:289:9)
    at Socket.Readable.push (node:internal/streams/readable:228:10)
    at TCP.onStreamRead (node:internal/stream_base_commons:190:23) {
  length: 240,
  severity: 'ERROR',
  code: '22P04',
  detail: undefined,
  hint: undefined,
  position: undefined,
  internalPosition: undefined,
  internalQuery: undefined,
  where: 'COPY temp_raw_tsv, line 470545: "470693\t2022-08-20 14:13:14.982328+00\t/new_block\t{"events": [{"txid": "0x5559844aefb5d7e7e28d45dc05ea..."',
  schema: undefined,
  table: undefined,
  column: undefined,
  dataType: undefined,
  constraint: undefined,
  file: 'copyfromparse.c',
  line: '839',
  routine: 'NextCopyFrom'
}
import-events process failed
```

**Comment by rafaelcr** [comment]
This error log looks to be related to the failing migration runs:
```
{"level":"error","message":"Error running pg-migrate Another migration is already running","stack":"Error: Another migration is already running\n    at lock (/app/node_modules/node-pg-migrate/dist/runner.js:41:15)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at exports.default (/a
pp/node_modules/node-pg-migrate/dist/runner.js:138:13)\n    at runMigrations (/app/src/datastore/migrations.ts:43:5)\n    at Function.connect (/app/src/datastore/pg-write-store.ts:123:7)\n    at init (/app/src/index.ts:120:24)","timestamp":"2022-10-12T16:28:35.789Z"}
{"level":"error","message":"app failed to start: Error: Another migration is already running Another migration is already running","stack":"Error: Another migration is already running\n    at lock (/app/node_modules/node-pg-migrate/dist/runner.js:41:15)\n    at processTicksAndRejections (node:internal/process/task_
queues:96:5)\n    at exports.default (/app/node_modules/node-pg-migrate/dist/runner.js:138:13)\n    at runMigrations (/app/src/datastore/migrations.ts:43:5)\n    at Function.connect (/app/src/datastore/pg-write-store.ts:123:7)\n    at init (/app/src/index.ts:120:24)","timestamp":"2022-10-12T16:28:35.793Z"}
```

**Comment by wileyj** [comment]
#1371

**Comment by wileyj** [comment]
> @wileyj I verified the fix is working for a v6 upgrade (in #1371), however, a v5 upgrade could still have some issues.
> 
> Do you think we can close this given that v6 addresses this problem? Or do you think a backported fix to v5 is warranted for this?

If i understand this correctly, #1371 would allow someone using an older version of the API to successfully migrate to v6?
If that's true, then ok to close - no reason to backport to v5 since i don't think there is a use-case for not using the latest release (i also have not encountered anyone requiring v5, usually they want to upgrade to the latest)

**Comment by rafaelcr** [comment]
> If i understand this correctly, https://github.com/hirosystems/stacks-blockchain-api/issues/1371 would allow someone using an older version of the API to successfully migrate to v6?

That's correct

> If that's true, then ok to close - no reason to backport to v5

Sounds great, thanks
